cmd/ocm/create/cluster/cmd.go

```diff
@@ -622,15 +622,19 @@ func promptAutoscaling(fs *pflag.FlagSet) error {
 		return err
 	}
 	if args.autoscaling.Enabled {
-		// set default
-		args.autoscaling.MinReplicas = minComputeNodes(args.ccs.Enabled, args.multiAZ)
+		// set default for interactive mode
+		if args.interactive {
```

Comment:
Is the problem that the flag should have no default at all when not interactive, or that you were overriding the flag value even when passed from command-line?

Some tips about implementation: (a bit subtle, so want to confirm desired behavior ^^ before I comment on implementation)

- `fs.Changed("min-replicas")` can tell you whether flag was passed from command line.
- _when not Changed()_, writing directly to a flag variable `args.autoscaling.MinReplicas = ...` effectively sets the default, which is useful for interactive prompts.
- when Changed(), interactive code should not touch the value.
- `fs.Set("min-replicas", "13")` would both change the variable and mark it Changed, so prompt will be skipped.  Don't want that.

Reply:
So it's both.
It not interactive I want to force the user to provide it.
And If it's interactive I want to cgane the default (that is 0 when non set) to the minimum allowed.

Reply:
I think this code still overrides command line values when --interactive is given :-1:
It is a feature that you can provide any subset of flags with --interactive, and these will be used without prompting.

One simple fix is to change the condition to `if args.autoscaling.MinReplicas == 0` or `if !fs.Changed("min-replicas") {`.
**That would also apply the minComputeNodes and MinReplicas defaults when not interactive.**  
If you go this way, you can also drop required checks from CheckIgnoredAutoscalingFlags.

Reply:
Ah, only now saw your reply.
> If not interactive I want to force the user to provide it.

In that case, I think you want `if args.interactive && args.autoscaling.MinReplicas == 0 {`.

Reply:
fixed